### DISCUSSION TITLE
Vincular la tarjeta almacenada a una orden generada.

### DISCUSSION THREAD

**Discussion by gonbooster** [creagia/laravel-redsys#37] [Ideas]
¡Hola!

Estaba reflexionando sobre la posibilidad de vincular una tarjeta a las solicitudes de redsys_requests en vuestra librería. 
Actualmente se almacenan pero no se vinculan a la orden .

Creo que sería beneficioso tener la capacidad de rastrear qué tarjeta se utilizó para realizar un pago, ya que esto podría ser valioso para la trazabilidad y podría tener aplicaciones prácticas más allá de este contexto.

¿Sería factible incorporar esta funcionalidad a la librería? Estoy interesado en conocer tu opinión y orientación al respecto.

Saludos,

**Comment by dtorras** [comment]
Podrías poner algún caso de ejemplo en que te parece interesante?

Según está implementado ahora, puedes relacionar la tarjeta tokenizada a un modelo Eloquent para poder volver a usarla después. El caso más típico es el de relacionar la tarjeta con el usuario para poder hacer pagos recursivos como una suscripción.

**Comment by gonbooster** [comment]
La situación principal sería determinar qué tarjeta se utilizó en cada orden. En caso de cualquier problema con la orden, sería crucial tener un registro de la tarjeta asociada.

En situaciones donde el sistema pueda fallar, la información relativa a la tarjeta podría resultar esencial al comunicarse con el banco correspondiente. 

Esta información podría ser necesaria para procesar devoluciones o realizar pagos adicionales en casos excepcionales.

 Mantener un seguimiento detallado de las tarjetas asociadas a cada orden contribuiría a una gestión más efectiva y transparente de cualquier contratiempo que pueda surgir durante el proceso.
